### DISCUSSION TITLE
Problems with Docker + NPM + Nging + Authelia [SOLVED]

### DISCUSSION THREAD

**Discussion by taimadoCE** [authelia/authelia#4283] [Q&A] [answered]
I'm having a problem with my conf and don't find solution to fix it.
My conf is based in Docker + NPM (Nginx Proxy Manarger) + Nginx + Authelia
All are installed and apparently fine. I configure the NPM to redirect my duckdns DDNS to my Nginx server.
All the Authelia conf was made following tutorials and videotutorials like this:

https://www.youtube.com/watch?v=4UKOh3ssQSU

All seems to work but when I try to logging in the web I get this:

![Screenshot 2022-10-27 at 13-00-44 Portainer local](https://user-images.githubusercontent.com/57089805/198268131-d8af1f62-3b21-4be1-a4af-186f292203a1.png)

And from "outside":
![Screenshot 2022-10-27 at 13-30-42 Portainer local](https://user-images.githubusercontent.com/57089805/198274063-4dec6c91-5785-4555-96c4-d61dc907aab7.png)

It seems like a loop. User is ok, pass is ok but the only thing I get is the auth authelia web again and again...

My NPM conf:
![imagen](https://user-images.githubusercontent.com/57089805/198269150-d72d84c2-e5b6-4627-bef8-9a2c241b6502.png)
Those ports are mapped in the router to mi local services.

**Comment by james-d-elliott** [comment]
Show your authelia config, also I recommend our guide for NPM: https://www.authelia.com/integration/proxies/nginx-proxy-manager/

**Reply by taimadoCE** [reply]
# yamllint disable rule:comments-indentation
---
###############################################################################
#                           Authelia Configuration                            #
###############################################################################

theme: dark #light/dark
jwt_secret: 1234567890abcdefghifjkl #any text or number you want to add here to create jwt Token

default_redirection_url: https://google.com/ #where to redirect for a non-existent URL

server:
  host: 0.0.0.0
  port: 9091
  

log:
  level: debug

totp:
  issuer: duckdns.org #your authelia top-level domain
  period: 30
  skew: 1

authentication_backend:
  disable_reset_password: false
  refresh_interval: 5m
  file:
    path: /config/users_database.yml #this is where your authorized users are stored
    password:
      algorithm: argon2id
      iterations: 1
      key_length: 32
      salt_length: 16
      memory: 1024
      parallelism: 8

access_control:
  default_policy: deny
  rules:
    ## bypass rule
    - domain: "authxxxxx.duckdns.org" #example domain to protect
      policy: bypass
    - domain: "xxxxx.duckdns.org" #example domain to protect
      policy: one_factor

      #add or remove additional subdomains as necessary. currenlty only supports ONE top-level domain
      #any time you add a new subdomain, you will need to restart the Authelia container to recognize the new settings/rules

session:
  name: authelia_session
  secret: fdgdfgsaudghxeuigftwehhfrincuusincrugtklsd #any text or number you want to add here to create jwt Token
  expiration: 3600  # 1 hour
  inactivity: 300  # 5 minutes
  domain: duckdns.org  # Should match whatever your root protected domain is

regulation:
  max_retries: 3
  find_time: 10m
  ban_time: 12h

storage:
  local:
    path: /config/db.sqlite3 #this is your database. You could use a mysql database if you wanted, but we're going to use this one.
  encryption_key: you_must_generate_a_random_string_of_more_than_twenty_chars_and_configure_this #added Dec 5 2021
  
notifier:
  disable_startup_check: true #true/false
  smtp:
    username: youremail@gmail.com #your email address
    password: Y0uRp@55W0rD! #your email password
    host: smtp.gmail.com #email smtp server
    port: 587 #email smtp port
    sender: youremail@gmail.com
    identifier: localhost
    subject: "[Authelia] {title}" #email subject
    startup_check_address: youremail@gmail.com
    disable_require_tls: false
    disable_html_emails: false
    tls:
      skip_verify: false
      minimum_version: TLS1.2
...

**Comment by james-d-elliott** [comment, marked as answer]
Adjust your session domain to the domain assigned to you via duckdns.org, i.e. `whatdemocracy.duckdns.org`, you can't use `duckdns.org` as browsers prevent cookies assigned to this domain (deletes them).

**Reply by taimadoCE** [reply]
Sorry but I must doing something wrong. I'm still having problems. I think that the problem is that I use two domains: the auth domain xxxx.duckdns.org and the main domain to protect yyyy.duckdns.org. Can be that the problem?

# yamllint disable rule:comments-indentation
---
###############################################################################
#                           Authelia Configuration                            #
###############################################################################

theme: dark #light/dark
jwt_secret: 1234567890abcdefghifjkl #any text or number you want to add here to create jwt Token

default_redirection_url: https://google.com/ #where to redirect for a non-existent URL

server:
  host: 0.0.0.0
  port: 9091
  

log:
  level: debug

totp:
  issuer: yyyy.duckdns.org #your authelia top-level domain
  period: 30
  skew: 1

authentication_backend:
  disable_reset_password: false
  refresh_interval: 5m
  file:
    path: /config/users_database.yml #this is where your authorized users are stored
    password:
      algorithm: argon2id
      iterations: 1
      key_length: 32
      salt_length: 16
      memory: 1024
      parallelism: 8

access_control:
  default_policy: deny
  rules:
    ## bypass rule
    - domain: "xxxx.duckdns.org" #example domain to protect
      policy: bypass
    - domain: "yyyy.duckdns.org" #example domain to protect
      policy: one_factor

      #add or remove additional subdomains as necessary. currenlty only supports ONE top-level domain
      #any time you add a new subdomain, you will need to restart the Authelia container to recognize the new settings/rules

session:
  name: authelia_session
  secret: fdgdfgsaudghxeuigftwehhfrincuusincrugtklsd #any text or number you want to add here to create jwt Token
  expiration: 3600  # 1 hour
  inactivity: 300  # 5 minutes
  domain: yyyy.duckdns.org  # Should match whatever your root protected domain is

regulation:
  max_retries: 3
  find_time: 10m
  ban_time: 12h

storage:
  local:
    path: /config/db.sqlite3 #this is your databse. You could use a mysql database if you wanted, but we're going to use this one.
  encryption_key: you_must_generate_a_random_string_of_more_than_twenty_chars_and_configure_this #added Dec 5 2021
  
notifier:
  disable_startup_check: true #true/false
  smtp:
    username: youremail@gmail.com #your email address
    password: Y0uRp@55W0rD! #your email password
    host: smtp.gmail.com #email smtp server
    port: 587 #email smtp port
    sender: youremail@gmail.com
    identifier: localhost
    subject: "[Authelia] {title}" #email subject
    startup_check_address: youremail@gmail.com
    disable_require_tls: false
    disable_html_emails: false
    tls:
      skip_verify: false
      minimum_version: TLS1.2
...

**Reply by james-d-elliott** [reply]
Correct, we can't protect multiple domains at this stage so you'll need two authelia containers. Just like you can't set a cookie for `.com` you can't set one for `.duckdns.org` either. In the next version we're implementing multidomain support but the SSO capabilities will not exist (you'll just have to sign in once per configured domain when you visit).

**Reply by taimadoCE** [reply]
First of all, thank you for your answers and your time. I'm learning step by step reading and researching by myself.
So as I understand reading your previous post to implement Authelia with duckdns domains the best option is to use one container per domain, isn't it?
Thanks for all.

**Reply by james-d-elliott** [reply]
Per duckdns domain that is correct. Unless you use `b.a.duckdns.org` and `c.a.duckdns.org`. for apps, in which case you'd use `a.duckdns.org` for the domain.

**Reply by taimadoCE** [reply]
Thanks again. I'll try that this aftermoon .

**Reply by taimadoCE** [reply]
> Per duckdns domain that is correct. Unless you use `b.a.duckdns.org` and `c.a.duckdns.org`. for apps, in which case you'd use `a.duckdns.org` for the domain.

It works!
Thanks a lot. The next step is combine it wit Fail2ban. ;-)
